src/lib/pages/backoffice/Patron/PatronDetails/ItemsSearch/ItemsSearch.js

```diff
@@ -16,27 +16,29 @@ export default class ItemsSearch extends Component {
     // but we need altered behaviour for the paste action
     // and in this way the state from before update is preserved
     // eslint-disable-next-line react/no-unused-state
-    this.state = { prevSearchQuery: '' };
+    this.state = { prevSearchQuery: '', executedSearch: false };
   }
 
-  onInputChange = queryString => {
-    const { updateQueryString } = this.props;
-    updateQueryString(queryString);
+  componentWillUnmount = () => {
+    const { clearResults } = this.props;
+    clearResults();
   };
 
   executeSearch = queryString => {
     const { queryString: propsQueryString, fetchItems } = this.props;
     queryString = queryString || propsQueryString;
     // eslint-disable-next-line react/no-unused-state
-    this.setState({ prevSearchQuery: queryString });
-    return fetchItems(queryString);
+    this.setState({ prevSearchQuery: queryString, executedSearch: true });
+    fetchItems(queryString);
   };
 
   onPasteHandler = async event => {
     const { checkoutItem, patronDetails } = this.props;
+    const { prevSearchQuery } = this.state;
     let queryString = event.clipboardData.getData('Text');
+    let doublePaste = prevSearchQuery === queryString;
```

Comment:
```suggestion
    const sameQueryString = prevSearchQuery === queryString;
```
I would change `let` to `const` whenever possible and also the name of the var: it could be that there was no pasting event eventually, but typing the same query string